lib/foreman/renderer/scope/report.rb

```diff
@@ -61,6 +113,16 @@ def valid_yaml_type(cell)
             cell.to_s
           end
         end
+
+        def valid_json_type(cell)
+          if cell.is_a?(String) || [true, false].include?(cell) || cell.is_a?(Numeric) || cell.nil?
+            cell
+          elsif cell.is_a?(Enumerable)
+            cell.map { |item| valid_json_type(item) }
```

Comment:
```suggestion
            if cell.is_a?(Hash)
              cell.map { |item| valid_json_type(item) }.to_h
            else
              cell.map { |item| valid_json_type(item) }
            end
```
This should be better handled, because if there are nested `Hash`es then they are "unpacked" and thus become `Array`s, so e.g. 
```ruby
hash = {
  "key": "value",
  "key2: {
    "key3": "value3"
  }
}
```
becomes
```
hash = [
  [:key, "value"],
  [:key2, { "key3": "value3" }]
]
```

Reply:
great catch! I took your idea just wrote it without the duplication like this
```ruby
hashify = cell.is_a?(Hash)
cell.map! { |item| valid_json_type(item) }
cell = cell.to_h if hashify
cell
```

sadly we need `hashify` as `is_a?(Hash)` needs to be detected before we run map on it.